clients/web/src/views/widgets/DateTimeWidget.js

```diff
@@ -0,0 +1,228 @@
+ */
+girder.views.DateTimeWidget = girder.View.extend({
+
+    /**
+     * @param [settings.prefix='default'] Prefix for element IDs in case
+     *     multiple DateTimeWidgets are rendered simultaneously.
+     */
+    initialize: function (settings) {
+        this.prefix = settings.prefix || 'default';
+        this.dateTimeId = '#' + this.prefix + '-datetime';
+    },
+
+    render: function () {
+        this.$el.html(girder.templates.dateTimeWidget({
+            prefix: this.prefix
+        }));
+
+        this.$(this.dateTimeId).datetimepicker({
+            showClear: true,
+            showTodayButton: true,
+            useCurrent: 'day'
+        });
+
+        return this;
+    },
+
+    /**
+     * Set date/time. Argument can be an ISO 8601-formatted date string, a
+     * moment object, or a Date object. Call with null or the empty string to
+     * clear. When called with a string, the widget displays in local time.
+     *
+     * @param date Date/time to display.
+     */
+    setDate: function (date) {
+        var picker = this._picker();
+        if (_.isEmpty(date)) {
+            picker.clear();
+        } else if (_.isString(date)) {
+            var localDate = moment.utc(date).local();
+            picker.date(localDate);
+        } else {
+            picker.date(date);
+        }
+    },
+
+    /**
+     * Get the date/time as a moment object in UTC. Returns null if no date is
+     * set.
+     */
+    date: function () {
+        var picker = this._picker();
+        var date = picker.date();
+        if (date !== null) {
+            date = moment(date);
+            date.utc();
+        }
+        return date;
+    },
+
+    /**
+     * Convenience function to return the date/time as a string in UTC. Returns
+     * the empty string if no date is set.
+     */
+    dateString: function () {
+        var date = this.date();
+        if (date === null) {
+            return '';
+        }
+        return date.format();
+    },
+
+    /**
+     * Convenience function to access the datetimepicker on an element.
+     */
+    _picker: function () {
+        var picker = this.$(this.dateTimeId).data('DateTimePicker');
+        return picker;
+    }
+});
+
+/**
+ * This widget provides text input fields to specify a range of dates/times.
+ * That is, the first field specifies "from" and the second field specifies
+ * "to." The user chooses each date/time using a popup picker.
+ */
+girder.views.DateTimeRangeWidget = girder.View.extend({
+
+    /**
+     * @param [settings.prefix='default'] Prefix for element IDs in case
+     *     multiple DateTimeRangeWidgets are rendered simultaneously.
+     */
+    initialize: function (settings) {
+        this.prefix = settings.prefix || 'default';
+        this.dateTimeFromId = '#' + this.prefix + '-datetime-from';
+        this.dateTimeToId = '#' + this.prefix + '-datetime-to';
+    },
+
+    render: function () {
+        var view = this;
```

Comment:
This is a minor nit, but I've started trying to avoid these sorts of `var foo = this` workaround bindings and just using `_.bind` instead to set the value of `this` explicitly. Not sure if it's worth being prescriptive about, just feels a bit cleaner to me.